src/_fixErroneousExpansionMdx.test.ts

```diff
@@ -0,0 +1,484 @@
+            [Time].[HistoricalDates].[AsOfDate].[2021-12-10],
+            Descendants(
+              {
+                [Currency].[Currency].[ALL].[AllMember]
+              },
+              [Currency].[Currency].[Currency]
+            )
+          )
+        )
+      ) ON ROWS
+      FROM [EquityDerivativesCube]
+      CELL PROPERTIES VALUE, FORMATTED_VALUE, BACK_COLOR, FORE_COLOR, FONT_FLAGS`);
+
+    const updatedMdx = _fixErroneousExpansionMdx(mdx, cube);
+
+    expect(stringify(updatedMdx, { indent: true })).toMatchInlineSnapshot(`
+      "SELECT
+        NON EMPTY Hierarchize(
+          Crossjoin(
+            Hierarchize(
+              [Time].[HistoricalDates].[AsOfDate].Members
+            ),
+            Hierarchize(
+              DrilldownLevel(
+                [Currency].[Currency].[ALL].[AllMember]
+              )
+            )
+          )
+        ) ON ROWS
+        FROM [EquityDerivativesCube]
+        CELL PROPERTIES VALUE, FORMATTED_VALUE, BACK_COLOR, FORE_COLOR, FONT_FLAGS"
+    `);
+  });
+
+  it("removes the problematic parts of the Mdx of a widget where a user collapsed and then re-expanded 2 different members in ActiveUI 4 (collapsing all of them, then re-expanding all of them)", () => {
+    const mdx = parse<MdxSelect>(`SELECT
+      NON EMPTY Hierarchize(
+        Union(
+          Crossjoin(
+            Hierarchize(
+              Union(
+                Hierarchize(
+                  DrilldownLevel(
+                    [Currency].[Currency].[ALL].[AllMember]
+                  )
+                ),
+                [Currency].[Currency].[ALL].[AllMember].[EUR]
+              )
+            ),
+            Hierarchize(
+              DrilldownLevel(
+                [Geography].[City].[ALL].[AllMember]
+              )
+            )
+          ),
+          Crossjoin(
+            [Currency].[Currency].[ALL].[AllMember].[EUR],
+            Descendants(
+              {
+                [Geography].[City].[ALL].[AllMember]
+              },
+              [Geography].[City].[City]
+            )
+          ),
+          Crossjoin(
+            [Currency].[Currency].[ALL].[AllMember].[GBP],
+            Descendants(
+              {
+                [Geography].[City].[ALL].[AllMember]
+              },
+              [Geography].[City].[City]
+            )
+          )
+        )
+      ) ON ROWS
+      FROM [EquityDerivativesCube]
+      CELL PROPERTIES VALUE, FORMATTED_VALUE, BACK_COLOR, FORE_COLOR, FONT_FLAGS`);
+
+    const updatedMdx = _fixErroneousExpansionMdx(mdx, cube);
+
+    expect(stringify(updatedMdx, { indent: true })).toMatchInlineSnapshot(`
+      "SELECT
+        NON EMPTY Hierarchize(
+          Crossjoin(
+            Hierarchize(
+              Union(
+                Hierarchize(
+                  DrilldownLevel(
+                    [Currency].[Currency].[ALL].[AllMember]
+                  )
+                ),
+                [Currency].[Currency].[ALL].[AllMember].[EUR]
```

Comment:
This sanitized Mdx is still bad: ideally, this node should be removed too.

That being said, while being suboptimal, this Mdx does work with ActiveUI 5. See https://github.com/activeviam/activeui/pull/2056. So I do not think that we should make `activeui-migration` more complex in order to cater for it.